core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/InlineClassAwareCaller.kt

```diff
@@ -96,6 +93,10 @@ internal class InlineClassAwareCaller<out M : Member?>(
 
             descriptor.valueParameters.mapTo(kotlinParameterTypes, ValueParameterDescriptor::getType)
         }
+
+        // If the default argument is set, ((paramSize / Int.SIZE_BITS) + 1) masks and one marker are added to the end of the argument.
+        val extraArgumentsTail = (if (isDefault) (kotlinParameterTypes.size / Int.SIZE_BITS) + 2 else 0) +
```

Comment:
The correct formula is `(kotlinParameterTypes.size + Int.SIZE_BITS - 1) / Int.SIZE_BITS` which means to round the division result up to the nearest integer:
https://github.com/JetBrains/kotlin/blob/d61e7db937499b0a26efacb8c10c5f92ec512aa2/core/reflection.jvm/src/kotlin/reflect/jvm/internal/KDeclarationContainerImpl.kt#L236

With the current changes, `callBy` should fail for functions that take 32 * N parameters, for integer N. Please add a function with exactly 32 parameters to the test case.

Reply:
Thanks for the review.
Fixed and added tests.

[952ab0a](https://github.com/JetBrains/kotlin/pull/4779/commits/952ab0adb76e99e9cf2c8f68c39a830bd6f9d633)
[0ec01f6](https://github.com/JetBrains/kotlin/pull/4779/commits/0ec01f656a2b42cbcde55ed6d47a36dead86f46c)

By the way, would it be better to make this logic common?

Reply:
Well, it's just an integer division with rounding up, I'm not sure that it warrants commonization. :)